### ISSUE TITLE
[flang] Missing diagnostics for conflicting EXTERNAL/INTRINSIC and TARGET attributes

### ISSUE THREAD

**Issue by MattPD** [llvm/llvm-project#197830]
Flang silently accepts declarations that specify both `EXTERNAL` and `TARGET` (or `INTRINSIC` and `TARGET`) attributes on the same entity, which violates the Fortran standard.

**Reproducer:**
```fortran
subroutine test1()
  real, target, external :: f1
end subroutine

subroutine test2()
  integer, target, intrinsic :: abs
end subroutine
```

**Expected behavior:** An error diagnostic should be emitted. An entity with the `EXTERNAL` or `INTRINSIC` attribute is a procedure, not a variable, and the `TARGET` attribute requires the entity to be a variable (F2018 §8.5.17).

**Actual behavior:** Flang accepts both cases without any diagnostic.

**Compiler Explorer repros** (all but Flang reject):
- Flang (trunk), https://godbolt.org/z/j6W4nnqT8, silently accepts
- gfortran (trunk), https://godbolt.org/z/5T69Ec6zx, `Error: TARGET attribute conflicts with EXTERNAL attribute`
- Intel ifx, https://godbolt.org/z/9hG1zo8qj, `error #6406: Conflicting attributes or multiple declaration of name.`
- nvfortran 26.3, https://godbolt.org/z/GYn1x91a6, `NVFORTRAN-S-0134-Illegal attribute - conflict with external`
- Cray Fortran (CCE), `ftn-425: Attributes EXTERNAL and TARGET must not appear in the same attribute list.`
